### ISSUE TITLE
Add BatchNorm1d into the nn module

### ISSUE THREAD

**Issue by antimora** [burn-rs/burn#198]
### Feature description

Currently there is `BatchNorm2d` but `BatchNorm1d` module is missing. 

### Feature motivation

Lots of sound NN operations use 1D modules, such as Conv1D. So having `BatchNorm1d` is necessary for training. I attempted to use BatchNorm2d by reshaping the input data, however, I got `[1,1,1,1]` type of error. 

### (Optional) Suggest a Solution

It might be useful to come up (or refactor from the existing code) with a universal BatchNorm function, similar to PyTorch's: https://pytorch.org/docs/stable/generated/torch.nn.functional.batch_norm.html and use for all 3 different dimensions.
